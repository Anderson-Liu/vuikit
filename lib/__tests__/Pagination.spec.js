import Vue from 'vue'
import Vuikit from 'vuikit'
import {
  createVue,
  destroyVM,
  renderSnapshot
} from 'jest/util'
import waitForUpdate from 'jest/wait-for-update'

Vue.use(Vuikit)

describe('Pagination', () => {
  let vm
  afterEach(() => {
    destroyVM(vm)
  })

  fit('renders correctly', () => {
    vm = createVue(`
      <vk-pagination>
        <vk-pagination-first />
        <vk-pagination-prev />
        <vk-pagination-pages :active="1" :total="200" :limit="10" />
        <vk-pagination-next />
        <vk-pagination-last />
      </vk-pagination>
    `)
    expect(renderSnapshot(vm)).toMatchSnapshot()
    destroyVM(vm)

    vm = createVue(`
      <vk-pagination>
        <vk-pagination-first />
        <vk-pagination-prev />
        <vk-pagination-pages :active="10" :total="200" :limit="10" />
        <vk-pagination-next />
        <vk-pagination-last />
      </vk-pagination>
    `)
    expect(renderSnapshot(vm)).toMatchSnapshot()
    destroyVM(vm)

    vm = createVue(`
      <vk-pagination>
        <vk-pagination-first />
        <vk-pagination-prev />
        <vk-pagination-pages :active="20" :total="200" :limit="10" />
        <vk-pagination-next />
        <vk-pagination-last />
      </vk-pagination>
    `)
    expect(renderSnapshot(vm)).toMatchSnapshot()
    destroyVM(vm)

    vm = createVue(`
      <vk-pagination :limit="10" :active="10" :total="200">
        <vk-pagination-first />
        <vk-pagination-prev />
        <vk-pagination-next />
        <vk-pagination-last />
      </vk-pagination>
    `)
    expect(renderSnapshot(vm)).toMatchSnapshot()
    destroyVM(vm)
  })

  it.skip('triggers event change when a not active page is clicked', done => {
    vm = createVue(`
      <vk-pagination>
        <vk-pagination-prev />
        <vk-pagination-pages :active="20" :total="200" :limit="10" />
      </vk-pagination>
    `, true)

    const cb = jest.fn()
    const pagination = vm.$children[0]

    pagination.$on('change', cb)
    vm.$el.querySelector('li a').click()
    waitForUpdate(() => {
      const args = cb.mock.calls[0]
      const { page } = args[0]
      expect(cb).toHaveBeenCalled()
      expect(args).toHaveLength(1)
      expect(page).toEqual(20)
    }).then(done)
  })
})
