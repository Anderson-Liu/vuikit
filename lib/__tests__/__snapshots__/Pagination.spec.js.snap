exports[`Pagination renders correctly 1`] = `
<ul class="uk-pagination">
  <li><a><span uk-pagination-previous="true" class="uk-pagination-prev uk-icon"></span></a></li>
  <li><a><span uk-pagination-previous="true" class="uk-pagination-prev uk-icon"></span></a></li>
  <li class="uk-active"><span>1</span></li>
  <li><a>2</a></li>
  <li><a>3</a></li>
  <li><a>4</a></li>
  <li><a>5</a></li>
  <li><a>6</a></li>
  <li><a>7</a></li>
  <li><span>...</span></li>
  <li><a>20</a></li>
  <li><a><span uk-pagination-next="true" class="uk-pagination-next uk-icon"></span></a></li>
  <li><a><span uk-pagination-next="true" class="uk-pagination-next uk-icon"></span></a></li>
</ul>
`;

exports[`Pagination renders correctly 2`] = `
<ul class="uk-pagination">
  <li><a><span uk-pagination-previous="true" class="uk-pagination-prev uk-icon"></span></a></li>
  <li><a><span uk-pagination-previous="true" class="uk-pagination-prev uk-icon"></span></a></li>
  <li><a>1</a></li>
  <li><span>...</span></li>
  <li><a>7</a></li>
  <li><a>8</a></li>
  <li><a>9</a></li>
  <li class="uk-active"><span>10</span></li>
  <li><a>11</a></li>
  <li><a>12</a></li>
  <li><a>13</a></li>
  <li><span>...</span></li>
  <li><a>20</a></li>
  <li><a><span uk-pagination-next="true" class="uk-pagination-next uk-icon"></span></a></li>
  <li><a><span uk-pagination-next="true" class="uk-pagination-next uk-icon"></span></a></li>
</ul>
`;

exports[`Pagination renders correctly 3`] = `
<ul class="uk-pagination">
  <li><a><span uk-pagination-previous="true" class="uk-pagination-prev uk-icon"></span></a></li>
  <li><a><span uk-pagination-previous="true" class="uk-pagination-prev uk-icon"></span></a></li>
  <li><a>1</a></li>
  <li><span>...</span></li>
  <li><a>14</a></li>
  <li><a>15</a></li>
  <li><a>16</a></li>
  <li><a>17</a></li>
  <li><a>18</a></li>
  <li><a>19</a></li>
  <li class="uk-active"><span>20</span></li>
  <li><a><span uk-pagination-next="true" class="uk-pagination-next uk-icon"></span></a></li>
  <li><a><span uk-pagination-next="true" class="uk-pagination-next uk-icon"></span></a></li>
</ul>
`;

exports[`Pagination renders correctly 4`] = `
<ul class="uk-pagination">
  <li><a><span uk-pagination-previous="true" class="uk-pagination-prev uk-icon"></span></a></li>
  <li><a><span uk-pagination-previous="true" class="uk-pagination-prev uk-icon"></span></a></li>
  <li><a><span uk-pagination-next="true" class="uk-pagination-next uk-icon"></span></a></li>
  <li><a><span uk-pagination-next="true" class="uk-pagination-next uk-icon"></span></a></li>
</ul>
`;
